web/src/hashchange.js

```diff
@@ -341,11 +341,13 @@ function do_hashchange_overlay(old_hash) {
     }
 
     if (base === "settings") {
+        admin.build_page();
         settings.launch(section);
         return;
     }
 
     if (base === "organization") {
+        settings.build_page();
         admin.launch(section);
         return;
     }
```

Comment:
This really reads like you've got things mixed up, with the admin case calling the settings build_page and vice versa; posted a comment above about a proposal that seems like it'd fix this while making the code better in other ways too.